src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -531,14 +600,105 @@ final class LazyList[+A] private(private[this] var lazyState: () => LazyList.Sta
     if (scout.isEmpty) state
     else tail.takeRightState(scout.tail)
 
+  override def slice(from: Int, until: Int): LazyList[A] = take(until).drop(from)
+
   override def reverse: LazyList[A] = reverseOnto(LazyList.empty)
 
+  override protected def reversed: LazyList[A] = reverse
+
+  override def reverseIterator: Iterator[A] = reverse.iterator
+
+  @deprecated("Use .reverse.map(f) instead of .reverseMap(f)", "2.13.0")
+  override def reverseMap[B](f: A => B): LazyList[B] = reverse.map(f)
+
   // need contravariant type B to make the compiler happy - still returns LazyList[A]
   @tailrec
   private def reverseOnto[B >: A](tl: LazyList[B]): LazyList[B] =
     if (isEmpty) tl
     else tail.reverseOnto(newLL(sCons(head, tl)))
 
+  override def diff(that: collection.Seq[_ >: A]): LazyList[A] =
+    if (knownIsEmpty) LazyList.empty
+    else super.diff(that)
+
+  override def intersect(that: collection.Seq[_ >: A]): LazyList[A] =
+    if (knownIsEmpty) LazyList.empty
+    else super.intersect(that)
+
+  // TODO: override to detect cycles
+  override def distinct: LazyList[A] = super.distinct
+
+  // TODO: override to detect cycles
+  override def distinctBy[B](f: A => B): LazyList[A] = super.distinctBy(f)
+
+  override def endsWith[B >: A](that: collection.Iterable[B]): Boolean =
+    takeRight(that.size).startsWith(that)
```

Comment:
`==` instead of `startsWith`? The size has to match anyway

Reply:
Come to think, I don't think there's a test of the laziness of `==` either. That should be added.